### DISCUSSION TITLE
How to rollback optimizer step (i.e. reload best checkpoint) during training?

### DISCUSSION THREAD

**Discussion by xsys-technology** [Lightning-AI/pytorch-lightning#13093] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
During training, if the optimizer takes a really bad step that sends validation loss through the roof, how can one rollback the step and reload best model checkpoint (and reset logging/pbar results/metrics so that _val_loss_ remains accurate)? 

I imagine the preferred way to do this is via a callback function (e.g. '_on_train_epoch_start_') but I'm not sure how to do this properly/safely when the model is dispatched and sharded across gpu's.
